R/Quandldatatable.R

```diff
@@ -0,0 +1,75 @@
+#' Returns the bulk download url
+#'
+#' @details Set your \code{api_key} with \code{Quandl.api_key} function. For instructions on finding your api key go to \url{https://www.quandl.com/account/api}
+#'
+#' @param code Datatable code on Quandl specified as a string.
+#' @param paginate When set to TRUE, fetches up to 1,000,000 rows of data
+#' @param ... Additional named values that are interpreted as Quandl API parameters.
+#' @return Returns a data.frame.
+#' @seealso \code{\link{Quandl.api_key}}
+#' @examples \dontrun{
+#' Quandl.datatable('ZACKS/FC', paginate=TRUE)
+#' }
+#' @export
+Quandl.datatable <- function(code, paginate=FALSE, ...) {
+  path <- paste0("datatables/", code)
+  params <- list(...)
+
+  # make request for first page of data
+  json <- do.call(quandl.api, c(path=path, params))
+  datatable <- json$datatable
+  data <- datatable$data
+  # contains a list of names and corresponding types
+  columns <- datatable$columns
+  next_cursor_id <- json$meta$next_cursor_id
+  df <- as.data.frame(data, stringsAsFactors=FALSE)
+
+  # continue to make requests for data if paginate=TRUE and there is data
+  while (isTRUE(paginate) && !is.null(next_cursor_id)) {
+    params['qopts.cursor_id'] <- next_cursor_id
+    json <- do.call(quandl.api, c(path=path, params))
+    df_page <- as.data.frame(json$datatable$data, stringsAsFactors=FALSE)
+    df <- rbind(df, df_page)
+    next_cursor_id <- json$meta$next_cursor_id
+
+    # only fetch a maximum of 1,000,000 rows
+    if (nrow(df) >= quandl.datatable.max_rows() && !is.null(next_cursor_id)) {
+      warning(paste("This call returns a larger amount of data than Quandl.datatable() allows.",
```

Comment:
Please update the statement based on this: https://quandl.atlassian.net/browse/AP-1701